operator/pkg/watch/restricted_enqueue_request_for_owner.go

```diff
@@ -162,6 +162,10 @@ func (e *RestrictedEnqueueRequestForOwner) getOwnerReconcileRequestFromOwnerRefe
 			e.Log.Error(err, "error getting owner")
 		}
 
+		// In case component state can't be updated due to operator error, still possible enqueue by Generation update.
+		if componentOld.GetGeneration() != componentNew.GetGeneration() {
```

Comment:
A state update will not trigger a generation change, only a spec update. If you cannot update the state and the generation still changed, I think it is safer to just requeue instead of tracking the generation. We can discuss this I think

Reply:
removed